jib-gradle-plugin-extension-api/src/main/java/com/google/cloud/tools/jib/gradle/extension/JibGradlePluginExtension.java

```diff
@@ -29,13 +30,28 @@
  * the plugin is configured to load the extension class, the Jib plugin extension framework calls
  * the interface method of the class.
  */
-public interface JibGradlePluginExtension extends JibPluginExtension {
+public interface JibGradlePluginExtension<T> extends JibPluginExtension {
+
+  /**
+   * The type of an custom configuration defined by this extension. The configuration object is
+   * mapped from {@code <pluginExtensions><pluginExtension><configuration>}. If the extension does
+   * not wish to define a custom configuration, this method may return {@code Optional#empty()}.
+   *
+   * @return type of an extension-specific custom configuration
+   */
+  default Optional<Class<T>> getExtraConfigType() {
```

Comment:
I always thought default was designed to be used when extending an interface and maintain backward compatibility. It makes sense here conceptually, but I generally would avoid it?

Reply:
Agreed! Also updated comments.